cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -320,34 +327,87 @@ pub fn check_build_metadata_from_storepath(
 
     Ok(CheckedBuildMetadata {
         drv_path: drv_path.to_string(),
-        package: pkg.to_string(),
         name,
+        pname: build_result.pname.clone(),
         outputs,
         outputs_to_install,
         system,
-        description: None,
-        version: None,
+        version: Some(build_result.version.clone()),
         _private: (),
     })
 }
 
 /// Collect metadata needed for publishing that is obtained from the build output
 pub fn check_build_metadata(
-    env: &impl Environment,
+    flox: &Flox,
+    env_metadata: &CheckedEnvironmentMetadata,
+    env: &PathEnvironment,
+    builder: &impl ManifestBuilder,
     pkg: &str,
 ) -> Result<CheckedBuildMetadata, PublishError> {
-    // For now assume the build is successful, and present.
-    // Look for the output from the build at `results-<pkgname>`
-    // Note that the current builds only support a single output at that
-    // pre-defined path.  Later work will get structured results from the build
-    // process to feed this.
-
-    let result_dir = build_symlink_path(env, pkg)?;
-    let storepath = result_dir
-        .read_link()
-        .map_err(|e| PublishError::NonexistentOutputs(e.to_string()))?;
-
-    let metadata = check_build_metadata_from_storepath(pkg, &storepath.to_string_lossy())?;
+    // git clone into a temp directory
+    let clean_repo_path = tempfile::tempdir_in(flox.temp_dir.clone()).unwrap();
+    let _ = GitCommandProvider::clone_branch(
+        env.parent_path().unwrap(),
+        &clean_repo_path,
+        &env_metadata.build_repo_ref.rev,
+        false,
+    );
+
+    let dot_flox_path = CanonicalPath::new(clean_repo_path.path().join(".flox"))
+        .map_err(|err| EnvironmentError::DotFloxNotFound(err.path))?;
+    let mut clean_build_env =
+        PathEnvironment::open(flox, PathPointer::new(env.name()), dot_flox_path)?;
+
+    // Build the package and collect the outputs
+    let output_stream = builder
+        .build(
+            flox,
+            &clean_build_env.parent_path()?,
+            &clean_build_env.build(flox).unwrap(),
+            &clean_build_env
+                .rendered_env_links(flox)
+                .unwrap()
+                .development,
+            &[pkg.to_owned()],
+        )
+        .unwrap();
+
+    let mut output_stdout = String::new();
+    let mut output_stderr = String::new();
+    let mut output_build_results: Option<BuildResults> = None;
+    for message in output_stream {
+        match message {
+            build::Output::Success(build_results) => {
+                let _ = output_build_results.insert(build_results);
+            },
+            build::Output::Failure(_) => {
+                panic!("expected build to succeed");
+            },
+            build::Output::Stdout(line) => {
+                println!("stdout: {line}"); // To debug failing tests
```

Comment:
i get there being a potential reason to maintain stdout/err as is in flox build, but here I'd rather log those to info! or debug! (considering docker builds are info)